src/snowflake/snowpark/dataframe_writer.py

```diff
@@ -4,6 +4,7 @@
 
 import sys
 from typing import Any, Dict, List, Literal, Optional, Union, overload
+from typing_extensions import Unpack
```

Comment:
`Unpack` is supported starting with `typing_extensions==4.1.0`.
https://github.com/python/typing_extensions/blob/e3359a92257b7abcbb5d24a1f3a81c1b007defbc/CHANGELOG#L1-L4